src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp

```diff
@@ -480,7 +480,7 @@ namespace
                 mtl->Update(time, FOREVER);
 
             const int submtlcount = mtl->NumSubMtls();
-            if (submtlcount > 0)
+            if (mtl->IsMultiMtl() && submtlcount > 0)
```

Comment:
What was the impact of not doing this test?

Reply:
It takes submaterials (e.g. layer materials) and assignes them to separate faces, as if they are multli-sub materials.

Reply:
I'm not sure to understand. Was there a bug before you added this test?

Reply:
Any material with submaterials was treated as multli-sub material. So yes, it was a bug probably.

Reply:
I see!